examples/gno.land/r/demo/users/users.gno

```diff
@@ -236,12 +236,11 @@ func GetUserByAddressOrName(input users.AddressOrName) *users.User {
 }
 
 func Resolve(input users.AddressOrName) std.Address {
-	name, isName := input.GetName()
-	if !isName {
-		return std.Address(input) // TODO check validity
+	if user := GetUserByAddressOrName(input); user != nil {
```

Comment:
Your version makes sense, but the current `Resolve` function is purposefully expecting either an address or a nickname starting with `@`. While it may not be look like being an ideal solution, this behavior was added when we decided to also configure `marked` to recognize `@XXX` in the generated Markdown.

I suggest that we add another helper function and add a comment above `Resolve` to differentiate the two behaviors.

Reply:
More  context: https://github.com/gnolang/gno/pull/175